source/MapPanel.cpp

```diff
@@ -1240,9 +1244,12 @@ void MapPanel::DrawMissions()
 		bool blink = false;
 		if(mission.Deadline())
 		{
-			int days = min(5, mission.Deadline() - player.GetDate()) + 1;
+			int days = min(6, mission.Deadline() - player.GetDate() + 1);
 			if(days > 0)
+			{
+				days = max(1, days - distance.Days(mission.Destination()->GetSystem()));
 				blink = (step % (10 * days) > 5 * days);
+			}
 		}
```

Comment:
Can this just be moved to a helper function (it can probably go in the namespace) so the whole thing isn't repeated?
I'm thinking something like `bool BlinkMissionIndicator(const PlayerInfo &player, const Mission &mission, const DistanceMap &distance)`.